torch/_inductor/decomposition.py

```diff
@@ -269,6 +269,62 @@ def angle(x):
         return ret + nan
 
 
+@register_decomposition([aten.add])
+def add(x, y, **kwargs):
+    x_is_complex_tensor = torch.is_tensor(x) and x.is_complex()
+    y_is_complex_tensor = torch.is_tensor(y) and y.is_complex()
+    if not x_is_complex_tensor and not y_is_complex_tensor:
+        return NotImplemented
+    r = y.real
+    i = y.imag if y_is_complex_tensor else 0.0
+    alpha = kwargs.get("alpha")
+    if alpha is not None:
+        r *= alpha
+        i *= alpha
+    r = x.real + r
+    if x_is_complex_tensor:
+        i = x.imag + i
+    complex_type = x.dtype if x_is_complex_tensor else y.dtype
+    return (
+        torch.where(
+            torch.arange(2, device=x.device, dtype=torch.uint8) == 0,
+            r.unsqueeze(-1),
+            i.unsqueeze(-1),
+        )
+        .view(complex_type)
+        .squeeze(-1)
+    )
+
+
+@register_decomposition([aten.real])
+def real(self):
+    if not torch.is_tensor(self) or not self.is_complex():
+        return NotImplemented
+    assert self.is_complex(), "real should only be called on a complex tensor"
+    if self.dtype == torch.complex32:
+        return self.view(torch.float16)[..., ::2]
```

Comment:
What wrapper code do these views get mapped to?

I'm worried this aliasing handling will be incorrect.

Reply:
I should inline this code into the `add` function above and remove the function here.